### PR TITLE
TFL: Update detection_postprocess kernel

### PR DESCRIPTION
**Pull request by mansnils**
This change uses std::stable_sort instead of std::partial_sort for the case where std::partial_sort use the full range.

This is the case preventing bit-exactness between TFL and TFLM. Problem with std::partial_sort is that the order of equal elements is not guaranteed to be preserved. Potentially std::partial_sort should be totally replaced but leaving that for now.

Next step would be to add a custom stable sort to TFLM, as std::stable_sort uses heap.

This progress towards: https://github.com/tensorflow/tensorflow/issues/47158

### CODE REVIEW

**Comment by advaitjain** [comment]
Some more context (from the TFLM team chatting with @mansnils, @freddan80 and the CMSIS-NN team):

The CMSIS-NN team has CI to make sure that the TFLM and TfLite kernels are bit exact and the use of std::partial sort has been preventing that for the detection_postprocess kernel.

If switching the lite implementation to use std::stable_sort is acceptable, then there is a path for the TFLM implementation to also be made bit exact.

The purpose of this PR is to make sure that such a change is ok for the TfLite kernel before going down the path of making a corresponding (but more involved) change to the TFLM kernel.

File: tensorflow/lite/kernels/detection_postprocess.cc [hunk 1]
```diff
@@ -367,6 +367,13 @@ void DecreasingPartialArgSort(const float* values, int num_values,
       [&values](const int i, const int j) { return values[i] > values[j]; });
 }
 
+void DecreasingArgSort(const float* values, int num_values, int* indices) {
+  std::iota(indices, indices + num_values, 0);
+  std::stable_sort(
```
Comment (jdduke) [inline tensorflow/lite/kernels/detection_postprocess.cc:372]: Probably worth a brief comment here as to why we need/want stable_sort.

Comment (mansnils) [reply tensorflow/lite/kernels/detection_postprocess.cc:375]: Done

**Review by srjoglekar246 — APPROVED** [review]

**Review by srjoglekar246 — CHANGES_REQUESTED** [review]
I profiled these changes internally & the latency seems fine, there is no noticable change.

Will it be possible to add a test in detection_postprocess_test.cc to verify that sorting retains order?

**Comment by advaitjain** [comment]
> I profiled these changes internally & the latency seems fine, there is no noticable change.
> 
> Will it be possible to add a test in detection_postprocess_test.cc to verify that sorting retains order?

@mansnils, unfortunately this PR got merged before you likely had a chance to address this comment from @srjoglekar246. Can you please open a new PR with the requested test, if adding an isolated test case is feasible?

**Comment by mansnils** [comment]
@advaitjain @srjoglekar246 Yes I will open a new PR.

**Pull request merged**